server/api/db/sqldb/db.py

```diff
@@ -4151,28 +4152,32 @@ def _delete_tagged_object(
                 "Neither name nor key specified when deleting an object."
             )
 
-        object_id = None
         obj_name = name or key
-        if uid:
-            object_record = self._query(
-                session,
-                cls,
-                project=project,
-                uid=uid,
-                name=name,
-                key=key,
-                **kwargs,
-            ).one_or_none()
-            if object_record is None:
-                return None, None
-            object_id = object_record.id
-        elif tag and tag != "*":
-            tag_record = self._query(
-                session, cls.Tag, project=project, name=tag, obj_name=obj_name
-            ).one_or_none()
-            if tag_record is None:
-                return None, None
-            object_id = tag_record.obj_id
+
+        # try to find the object by given arguments
+        query = self._query(
+            session,
+            cls,
+            project=project,
+            uid=uid,
+            name=name,
+            key=key,
+            **kwargs,
+        )
+
+        # join on tags if given
+        if tag and tag != "*":
```

Comment:
And another thing here - conceptually, if someone asks for `*` tag it may mean that we're looking for all tagged objects (which is strange anyway, since we're looking a `one_or_none` so just a single object) . Conceptually, I would say that a `*` tag should result in a join with tags table, but without filtering on tag name, thus returning all tagged results. 
Have to admin I didn't really check where a `*` tag may come from, and if we accept it at all, just a concern I have based on looking specifically at this code.

Reply:
I suppose previously if no uid was passed and `tag=="*"` then we would fall to the last part that deletes all objects with the given key.

> Conceptually, I would say that a * tag should result in a join with tags table, but without filtering on tag name, thus returning all tagged results.

^ While this sounds legit, it does change the behavior